catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -211,34 +154,87 @@ public void setEnableBackupPlugin(boolean enablePlugin) {
      * @param dir absolute path for the root file system backup directory.
      */
     public void setRootBackupDir(String dir) {
-        if (StringUtils.isBlank(dir)) {
-            LOGGER.error("The root backup directory is blank.");
-            return;
+        Validate.notNull(dir);
+        File directory = new File(dir);
+        validateDirectory(directory);
+        this.rootBackupDir = directory;
+        LOGGER.trace("Set root backup directory to: {}", this.rootBackupDir.toString());
+    }
+
+    ExecutorService getExecutor() {
+        return executor;
+    }
+
+    void setExecutor(ExecutorService executor) {
+        this.executor = executor;
+    }
+
+    private synchronized void execute(Runnable task) {
```

Comment:
If different create or delete jobs are hitting the backup plugin at nearly the same time, I need to preserve the order in which those calls happened.